### PR TITLE
Added deezer, SEARCH_SENSITIVITY option for metadata, refactored a bit and fixed some bugs

### PR DESCRIPTION
**Pull request by danimateo**
Yeah I know this is a kind of a lot, check it out and tell me what you think. I found deezer to be a better source than itunes for international songs, but the search sensitivity should be lowered a bit. At the moment it was at 1 (left it at this default), but that means there needs to be a perfect match between intersection and union in the search algo, which I think is too restrictive. Maybe we should lower it in the future, but for now it's configurable.

### CODE REVIEW

**Comment by deepjyoti30** [comment]
Hey @danimateo I would love to merge this PR.

However, I think there is no need for the `SEARCH_SENSITIVITY` tag as of now since we probably want the best matches of song data itself.

Can you please remove that option? Once you do that I'll go ahead and merge the PR.

Another thing, please try to follow [pep8](https://www.python.org/dev/peps/pep-0008/) for linting. You can install a linter. I have been meaning to add a ```Contribution.md``` but haven't had quite the time to do so.

**Comment by danimateo** [comment]
Hey @deepjyoti30 , I added the `SEARCH_SENSITIVITY` because Deezer doesn't seem to work that well if it's set to the default 1 value. The results for the other providers are still very good with it a bit lower. Can you test with some songs and tell me what you think?  If you still don't think it's alright as a configurable option, I'll remove it.

Regarding linting, I am using the default one VSC recommends, which is `autopep8`. I've noticed there are differences in spacing mostly, are you using something different or do you have a different config?

**Comment by deepjyoti30** [comment]
Hey @danimateo I cloned your branch and tried testing with some songs, weirdly enough no song was found by the API.

I then tried the endpoint myself with different songs and every time it just returned an empty array.

I even tried setting to 0.5 on the config.

Anything I'm doing wrong??

**Comment by danimateo** [comment]
Hey @deepjyoti30 can you tell me what song you searched for? Really weird

**Comment by deepjyoti30** [comment]
@danimateo I search for `Psycho`, the preferred result is the one by Post Malone. I also tried searching for `Blame it on me` and some Hindi songs like `Ilahi`.

Everytime it just returned an empty array.

**Comment by danimateo** [comment]
I tried searching with just deezer enabled and everything worked really good. Let me investigate some more, really strange behavior

**Comment by danimateo** [comment]
Ah @deepjyoti30 I think I got it. The API differentiates based on the country. Deezer isn't available in India unfortunately. Try it again with a proxy and it should work. Well, I don't think the option should be removed because of this since you can remove it in the config, but your call.

**Comment by deepjyoti30** [comment]
@danimateo If you're testing with ytmdl, remove itunes and other sources and only keep `deezer` in the config.

I mean the API is good. I think I can test it out using a proxy.

For the `SEARCH_SENSITIVITY` option, I think we can hardcode it to `0.5`?! I just don't want to add config options for each API, that way we should also support config options for other API's so I think that would probably be a big change.

**Comment by danimateo** [comment]
Check out #104, besides other refactoring I made it really easy to add a new provider should there be a need. And with the search sensitivity, I think it could work better as a separate config for all files because deezer tends to be more vague, as in long titles with other info included (sometimes), in contrast to itunes and gaana which are more curated and have short titles. I think 0.5 though is a sane default and has good results for all of them though. I'm inclined to keep it configurable though for now at least. Maybe when we have more options for each provider?

**Comment by deepjyoti30** [comment]
Hey @danimateo Just tested using a proxy. It's working good.

I understand that keeping the value configurable would be a nice option but for now let's keep it for later. Anyway most end users just want to use the tool out of the box so we need to keep a moderate value of `SEARCH_SENSITIVITY` as default.

I think we can add this API as well. Just change the config thing and I'll merge it.

We will surely make this configurable in a future release but for now let's not go with that. Also I checked the code that is handling the `SEARCH_SENSITIVITY`, I think we need to make some refactoring there as well but let's keep it for later. If you want you can also open an issue to discuss regarding this and we'll let users give their inputs on the same.

So once, you make the changes, ( please also fix the conflicts :sweat_smile: ), I'll merge it.

**Comment by danimateo** [comment]
@deepjyoti30 So you want the default to be 0.5 or hardcode it and remove it from settings?

**Comment by danimateo** [comment]
@deepjyoti30 I merged it, hope I didn't make any mistakes lol.

**Comment by deepjyoti30** [comment]
@danimateo Thanks, I'll take a look!

File: ytmdl/metadata.py [hunk 1]
```diff
@@ -136,29 +150,31 @@ def SEARCH_SONG(q="Tera Buzz", filters=[]):
     GET_METADATA_ACTIONS = {
         'itunes': get_from_itunes,
         'gaana': get_from_gaana,
+        'deezer': get_from_deezer
```
Comment (deepjyoti30) [inline ytmdl/metadata.py:153]: Hey @danimateo I think this line will throw an error?

**Comment by deepjyoti30** [comment]
Hey @danimateo I have merged this PR, I have made some changes in the code before merging.

This was mainly because since now we have 4 sources, I though we need to create a seperate directory for the metadata sources.

Also while going through your commits I saw that you had added a default `GET_EXTRA_DATA` dict in the `DEFAULTS` class and for that you had to import the `deezer` module. However, I think that is not a good practice since the `defaults` module is for all the default values of values that are configurable from the `config`.

So I moved those config options to a different file inside the `meta` directory, named `preconfig` which has a class named `CONFIG`.

Basically this class works the same as the `DEFAULTS` class however this class is for all the internal configurations related to the `metadata` sources.

Consider taking a look at the commits.

Also, if possible, please check if the `deezer` source is working properly after the changes I made since I cannot test it.
